src/qibo/tests/test_noise.py

```diff
@@ -36,24 +36,21 @@ def test_pauli_error(backend, density_matrix, nshots):
     target_circuit.add(gates.M(0, 1, 2))
 
     initial_psi = random_density_matrix(3) if density_matrix else random_state(3)
-    np.random.seed(123)
-    K.set_seed(123)
-    final_state = noise.apply(circuit)(initial_state=np.copy(initial_psi), nshots=nshots)
+    backend.set_seed(123)
+    final_state = backend.execute_circuit(noise.apply(circuit), initial_state=np.copy(initial_psi), nshots=nshots)
     final_state_samples = final_state.samples() if nshots else None
-    np.random.seed(123)
-    K.set_seed(123)
-    target_final_state = target_circuit(initial_state=np.copy(initial_psi), nshots=nshots)
+    backend.set_seed(123)
+    target_final_state = backend.execute_circuit(target_circuit, initial_state=np.copy(initial_psi), nshots=nshots)
     target_final_state_samples = target_final_state.samples() if nshots else None
 
     if nshots is None:
-        K.assert_allclose(final_state, target_final_state)
+        backend.assert_allclose(final_state, target_final_state)
     else:
-        K.assert_allclose(final_state_samples, target_final_state_samples)
+        backend.assert_allclose(final_state_samples, target_final_state_samples)
 
 
 @pytest.mark.parametrize("density_matrix", [False, True])
```

Comment:
Since the `ThermalRelaxationChannel` is working only with density matrices we should only test with the flag `density_matrix=True`. Similarly for the `ResetChannel`.